CRM/Member/WorkflowMessage/Membership/Membership.php

```diff
@@ -145,10 +165,23 @@ private function addExampleData(GenericWorkflowMessage $messageTemplate, array $
    * @throws \CRM_Core_Exception
    */
   private function getPriceSet(): ?array {
-    return (array) PriceSet::get(FALSE)
-      ->addWhere('extends', '=', CRM_Core_Component::getComponentID('CiviMember'))
-      ->addOrderBy('is_quick_config', 'DESC')
-      ->execute()->indexBy('id');
+    if (!$this->priceSets) {
+      $priceSets = (array) PriceSet::get(FALSE)
+        ->addWhere('extends', '=', CRM_Core_Component::getComponentID('CiviMember'))
```

Comment:
Something is quite wrong here. According to the schema, `extends` is a serialized field
https://github.com/civicrm/civicrm-core/blob/45e24b720576413c2fa4e4ccf6ae9fa6b3cfeb4e/xml/schema/Price/PriceSet.xml#L136-L146

This clause should fail to find any records because `=` is incompatible with serialized-bookend. The sql that this `=` clause generates is `WHERE (`a`.`extends` = \"3\")`. But when I test it in the API Explorer, it *does* find some records! I don't understand why that is, the sample data appears to be serialized properly.

Regardless, I think this line ought to be 
```suggestion
        ->addWhere('extends:name', 'CONTAINS', 'CiviMember')
```

which also works for me in the explorer.